quantlib/termstructures/volatility/equityfx/black_variance_curve.pyx

```diff
@@ -0,0 +1,91 @@
+include '../../../types.pxi'
+from cython.operator cimport dereference as deref
+
+from libcpp cimport bool
+from libcpp.vector cimport vector
+
+from quantlib.handle cimport shared_ptr
+from quantlib.time.date cimport Date, date_from_qldate
+from quantlib.time._date cimport Date as _Date
+from quantlib.time.daycounter cimport DayCounter
+from quantlib.time._daycounter cimport DayCounter as QlDayCounter
+
+from .black_vol_term_structure cimport BlackVarianceTermStructure
+from . cimport _black_vol_term_structure as _bvts
+from . cimport _black_variance_curve as _bvc
+from ..._vol_term_structure cimport VolatilityTermStructure
+
+
+cdef inline _bvc.BlackVarianceCurve* get_bvc(BlackVarianceCurve bvc):
+    """ Utility function to extract a properly casted BlackVarianceCurve out
+    of the internal _thisptr attribute of the BlackVolTermStructure base class.
+    """
+    cdef _bvc.BlackVarianceCurve* ref = <_bvc.BlackVarianceCurve*>bvc._thisptr.get()
+    return ref
+
+
+cdef class BlackVarianceCurve(BlackVarianceTermStructure):
+    """
+    Black volatility curve modelled as variance curve
+    This class calculates time-dependent Black volatilities using
+    as input a vector of (ATM) Black volatilities observed in the
+    market.
+
+    The calculation is performed interpolating on the variance curve.
+    Linear interpolation is used as default; this can be changed
+    by the setInterpolation() method.
+
+    For strike dependence, see BlackVarianceSurface.
+
+    todo check time extrapolation
+
+
+    Attributes
+    ----------
+    reference_date : Date
+    dates : list of :obj:`Date`
+    black_vols : list of :obj:`Volatility`
+    day_counter: DayCounter
+    force_monotone_variance: bool
+
+    """
+
+    def __init__(self,
+                 Date reference_date,
+                 dates,
+                 vector[Volatility] black_vols,
+                 DayCounter day_counter,
+                 bool force_monotone_variance = True,
+                 ):
+
+        cdef vector[_Date] _dates
+        for d in dates:
+            _dates.push_back(deref((<Date?>d)._thisptr))
+
+        self._thisptr = shared_ptr[VolatilityTermStructure](
+                                   new _bvc.BlackVarianceCurve(
+                                               deref(reference_date._thisptr),
+                                               _dates,
+                                               black_vols,
+                                               deref(day_counter._thisptr),
+                                               force_monotone_variance,
+                                               ))
+    # TermStructure interface
```

Comment:
Commented code should be removed or uncommented. Can you check?

Reply:
removed in 1244cdc